actions/class.Runner.php

```diff
@@ -380,62 +393,34 @@ public function getItem()
     {
         $code = 200;
 
-        $itemRef        = $this->getRequestParameter('itemDefinition');
-        $itemIdentifier = $this->getRequestParameter('itemIdentifier');
+        $itemIdentifier = $this->getRequestParameter('itemDefinition');
 
         try {
             $serviceContext = $this->getServiceContext(false, true);
 
-            //load item data
-            $response = $this->getItemDataResponse($itemRef, $itemIdentifier);
-
-            //start the timer
-            $this->runnerService->startTimer($serviceContext);
-
-        } catch (common_Exception $e) {
-            $response = $this->getErrorResponse($e);
-            $code = $this->getErrorCode($e);
-        }
-
-        $this->returnJson($response, $code);
-    }
-
-    /**
-     * Provides the definition data and the state of the
-     * next item.
-     */
-    public function getNextItemData()
-    {
-        $code = 200;
-
-        try {
-
-            $allowed = $this->runnerService->getTestConfig()->getConfigValue('allowBrowseNextItem');
-            if(!$allowed){
-                \common_Logger::w("Attempt to disclose the next item whitout the configuration");
-                throw new \common_exception_Unauthorized();
-            }
-            $serviceContext = $this->getServiceContext();
-            $session        = $serviceContext->getTestSession();
-            $route          = $session->getRoute();
-
-            if(!$route->isLast()){
-
-                $assessmentItemRef = $route->getNext()->getAssessmentItemRef();
-                $itemDefinition    = $assessmentItemRef->getHref();
-                $itemIdentifier    = $assessmentItemRef->getIdentifier();
-
-                $response = $this->getItemDataResponse($itemDefinition, $itemIdentifier);
-
-                if(is_array($response)){
-                    $response['itemDefinition'] = $itemDefinition;
+            if (is_array($itemIdentifier)) {
+                if (!$this->runnerService->getTestConfig()->getConfigValue('itemCaching.enabled')) {
+                    \common_Logger::w("Attempt to disclose the next items without the configuration");
+                    throw new \common_exception_Unauthorized();
+                }
+                
+                $response = [];
+                foreach ($itemIdentifier as $itemId) {
+                    //load item data
+                    $response['items'][] = $this->getItemData($itemId);
                 }
             } else {
-                $response = [
-                    'success' => true
-                ];
+                //load item data
+                $response = $this->getItemData($itemIdentifier);
+            }
+
+            if (is_array($response)) {
+                $response['success'] = true;
             }
 
+            //start the timer
+            $this->runnerService->startTimer($serviceContext);
```

Comment:
You shouldn't start the timer if you retrieve items for caching. In the other actions there's a `start` parameter.